src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKYamlDSLParser.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cameltooling.lsp.internal.parser;
+
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.TextDocumentItem;
+
+import com.github.cameltooling.lsp.internal.instancemodel.CamelURIInstance;
+import com.github.cameltooling.lsp.internal.instancemodel.YamlDSLModelHelper;
+
+/**
+ * @author Lars Heinemann
+ */
+public class CamelKYamlDSLParser extends ParserFileHelper {
+	
+	public static final String STEP_KEY = "steps:";
+	public static final String URI_KEY = "uri:";
+	public static final String REST_KEY = "rest:";
+	public static final String FROM_KEY = "from:";
+	public static final String TO_KEY = "- to:";
+
+	@Override
+	public String getCamelComponentUri(String line, int characterPosition) {
+		String camelComponentURI = null;
+		String trLine = line.trim();
+		if (trLine.startsWith(URI_KEY) && URI_KEY.length() < characterPosition) {
+			camelComponentURI = extractCamelComponentURI(trLine, URI_KEY);
+		} else if(trLine.startsWith(FROM_KEY) && trLine.indexOf('"') >= trLine.indexOf(':') + 1) {
+			camelComponentURI = extractCamelComponentURI(trLine, FROM_KEY);
```

Comment:
now that there is the test with the from, I notice that this line is not used. In fact, This case cannot happen. For the from:, there is always the "uri:"
Removing the line will simplify code

Reply:
I am not sure I can follow your thoughts. Can u elaborate a bit more how you come to this conclusion?

Reply:
you added a test testign completion for the completion for a "from".
i noticed that this line of code is still not covered. So I investigated why.
In fact, for the "from", there will always be a "uri" attribute too. Otherwise the yaml route cannot be valid.

The potential shortcut that we have for the "to" is not possible because we have at least a to after the "steps" inside it.